app/src/main/java/com/xmartlabs/scasas/criminalintent/ui/crime/list/CrimeListFragment.java

```diff
@@ -30,15 +32,31 @@
   public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
     View view = inflater.inflate(R.layout.fragment_crime_list, container, false);
     ButterKnife.bind(this, view);
+    setpuRecyclerView();
+
+    return view;
+  }
+
+  private void setpuRecyclerView() {
     crimeRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+    adapter = new CrimeAdapter(this::onCrimeTapped);
+    crimeRecyclerView.setAdapter(adapter);
```

Comment:
**ProTip:** you can do `crimeRecyclerView.setHasFixedSize(true)` to optimize its rendering by letting it know it won't change its size.